examples/nextjs-spa/pages/recovery.tsx

```diff
@@ -0,0 +1,126 @@
+// React
+import { useEffect, useState } from "react"
+import { useRouter } from "next/router"
+
+// Ory SDK
+import { ory } from "../components/sdk"
+import { RecoveryFlow, UpdateRecoveryFlowBody } from "@ory/client"
+
+import { AxiosError } from "axios"
+import type { NextPage } from "next"
+
+import { UserAuthCard } from "@ory/elements"
+import React from "react"
+import Link from "next/link"
+
+const Recovery: NextPage = () => {
+  const [flow, setFlow] = useState<RecoveryFlow>()
+
+  // Get ?flow=... from the URL
+  const router = useRouter()
+  const { flow: flowId, return_to: returnTo } = router.query
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return
+    }
+
+    // If ?flow=.. was in the URL, we fetch it
+    if (flowId) {
+      ory
+        .getRecoveryFlow({ id: String(flowId) })
+        .then(({ data }) => {
+          setFlow(data)
+        })
+        .catch((err: AxiosError) => {
+          switch (err.response?.status) {
+            case 410:
+            // Status code 410 means the request has expired - so let's load a fresh flow!
+            case 403:
+              // Status code 403 implies some other issue (e.g. CSRF) - let's reload!
+              return router.push("/recovery")
+          }
+
+          throw err
+        })
+      return
+    }
+
+    // Otherwise we initialize it
+    ory
+      .createBrowserRecoveryFlow()
+      .then(({ data }) => {
+        setFlow(data)
+      })
+      .catch(
+        (error: AxiosError) =>
+          error.response?.status === 404 && router.push("/login"),
+      )
+      .catch((err: AxiosError) => {
+        // If the previous handler did not catch the error it's most likely a form validation error
+        if (err.response?.status === 400) {
+          // Yup, it is!
+          setFlow(err.response?.data)
+          return
+        }
+
```

Comment:
can't we redirect the user to an error page or show them an error if it is something other than form validation? 

Like for instance the CSRF cookie could be missing and thus would fail submitting the form